os/include/tinyara/mpu.h

```diff
@@ -34,6 +34,18 @@
  * Pre-processor Definitions
  ********************************************************************************/
 
+/* Structure for MPU region numbers */
+
+struct mpu_regs {
+	uint8_t nmpu_regions;
+	uint8_t common_regions;
+	uint8_t app_regions;
+	uint8_t stackovf_regions;
+	uint8_t maxmpu_regions;
+};
+
+extern struct mpu_regs g_mpuregs;
```

Comment:
The name 'g_mpuregs' looks like "gloabl_mpu_registers". We already use `reg` in tcb_s for register. Let's rename it as `g_mpu_usages`? `g_mpu_region_info`? or something else..

Reply:
SAM recommends not to use global variable. Let's make a new function and call it from outside to get mpu region number information.
```
in mpu.h
enum mpu_region_usages {
   MPU_REGION_BOARD_SPECIFIC,
   ...
}

in up_mpuinit.c (common file)
static struct mpu_usages_s g_mpu_usages;

uint8_t mpu_get_region_info(enum mpu_region_usages usages)
{
     uint8_t nregion;

     switch (usages) {
     case MPU_REGION_BOARD_SPECIFIC:
          nregion = g_mpu_xxxx.nregion_board_specific;
           break;
      case ...
      }
   return nregion;
}
```

We need to make a set function for board specific value.

Reply:
modified